patches/server/0936-Call-missing-BlockDispenseEvent.patch

```diff
@@ -0,0 +1,87 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Sat, 29 Oct 2022 15:41:56 +0200
+Subject: [PATCH] Call missing BlockDispenseEvent
+
+
+diff --git a/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java b/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java
+index 9d601bf258a6d209a889afa7ba7f839c01f83cd8..f2f5b033893df21acffa0d39870022d5c2d6318c 100644
+--- a/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java
++++ b/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java
+@@ -1083,6 +1083,14 @@ public interface DispenseItemBehavior {
+                 BlockState iblockdata = worldserver.getBlockState(blockposition);
+ 
+                 this.setSuccess(true);
++
++                // Paper start
++                ItemStack result = org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockDispenseEvent(pointer, blockposition, stack, this);
++                if (result != null) {
++                    return result;
++                }
++                // Paper end
++
+                 if (iblockdata.is(Blocks.RESPAWN_ANCHOR)) {
+                     if ((Integer) iblockdata.getValue(RespawnAnchorBlock.CHARGE) != 4) {
+                         RespawnAnchorBlock.charge(worldserver, blockposition, iblockdata);
+@@ -1104,6 +1112,14 @@ public interface DispenseItemBehavior {
+                 BlockPos blockposition = pointer.getPos().relative((Direction) pointer.getBlockState().getValue(DispenserBlock.FACING));
+                 ServerLevel worldserver = pointer.getLevel();
+                 BlockState iblockdata = worldserver.getBlockState(blockposition);
++
++                // Paper start
++                ItemStack result = org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockDispenseEvent(pointer, blockposition, stack, this);
++                if (result != null) {
++                    return result;
++                }
++                // Paper end
++
+                 Optional<BlockState> optional = HoneycombItem.getWaxed(iblockdata);
+ 
+                 if (optional.isPresent()) {
+@@ -1129,6 +1145,13 @@ public interface DispenseItemBehavior {
+                     BlockPos blockposition = pointer.getPos();
+                     BlockPos blockposition1 = pointer.getPos().relative((Direction) pointer.getBlockState().getValue(DispenserBlock.FACING));
+ 
++                    // Paper start
++                    ItemStack result = org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockDispenseEvent(pointer, blockposition1, stack, this);
++                    if (result != null) {
++                        return result;
++                    }
++                    // Paper end
```

Comment:
This needs to be in the else clause of the conditional below otherwise it will create duplicate events firing. BlockDispenseEvent is also fired in the defaultDispenseItemBehavior#dispense method.